airflow/contrib/example_dags/example_azure_cosmosdb_sensor.py

```diff
@@ -41,24 +41,25 @@
     'email_on_retry': False
 }
 
-dag = DAG('example_azure_cosmosdb_sensor', default_args=default_args)
+with DAG(
+    dag_id='example_azure_cosmosdb_sensor',
+    default_args=default_args
+) as dag:
```

Comment:
This isn't going to work - the dag won't be found by Airflow as it needs to be stored in a module-level variable.

So it has to be something like this:
```python
dag = DAG('example_azure_cosmosdb_sensor', default_args=default_args)
with dag:
```

(also you removed the `dag.doc_md = __doc__` line)

Reply:
I followed GCP examples :
https://github.com/apache/airflow/blob/master/airflow/gcp/example_dags/example_cloud_sql.py#L186
https://github.com/apache/airflow/blob/master/airflow/gcp/example_dags/example_dataproc.py#L53

I removed `dag.doc` as it doesn't appear in all examples and I think it should be consistent.
There are many optional settings that can be defined but in my point of view example should be simple and to the point. If something isn't required for the behavior/usage you want to show then it shouldn't be there.

WDYT?

Reply:
Turns out this does work and I mis-understoon python's scoping rules.

Please keep the doc_md -- that will show up in the UI. It can be passed as an kwarg to `DAG()` though

Reply:
@ashb addressed comments